src/api/ccUtil.js

```diff
@@ -758,6 +775,244 @@ const ccUtil = {
     }
     return p;
   },
+
+  /**
+   * ---------------------------------------------------------------------------
+   * BTC APIs
+   * ---------------------------------------------------------------------------
+   */
+
+
+  /**
+   * Filter btc addresses by amount, return the addresses with sufficient amount. 
+   * @param addressList All the btc addresses.
+   * @param amount The amount to fit.
+   */
+  async filterBtcAddressByAmount(addressList, amount) {
+      let addressWithBalance = [];
+      for (let i = 0; i < addressList.length; i++) {
+          let utxos = await this.getBtcUtxo(config.MIN_CONFIRM_BLKS, config.MAX_CONFIRM_BLKS, [addressList[i].address]);
+
+          let result = await this.getUTXOSBalance(utxos);
+
+          addressWithBalance.push({
+              'address': addressList[i].address,
+              'balance': Number(web3.toBigNumber(result).div(100000000).toString())
+          });
+      }
+
+      addressWithBalance = addressWithBalance.sort((a, b) => {
+          return b.balance - a.balance;
+      });
+
+      let addressListReturn = [];
+      let totalBalance = 0;
+      for (let i = 0; i < addressWithBalance.length; i++) {
+          totalBalance += addressWithBalance[i].balance;
+          addressListReturn.push(addressWithBalance[i].address);
+
+          if(totalBalance > Number(amount)) {
+              break;
+          }
+      }
+
+      return addressListReturn;
+  },
+
+  /**
+   */
+  getUTXOSBalance(utxos) {
+      let sum = 0
+      let i = 0
+      for (i = 0; i < utxos.length; i++) {
+          sum += utxos[i].value
+      }
+      return sum
+  },
+
+  /**
+   */
+  async getBtcUtxo(minconf, maxconf, addresses) {
+      let utxos = await this._getBtcUtxo(minconf, maxconf, addresses);
+      let utxos2 = utxos.map(function (item, index) {
+          let av = item.value ? item.value : item.amount;
+          item.value = Number(web3.toBigNumber(av).mul(100000000));
```

Comment:
the value use 'Number()', should it be change to toBigNumber(), in case of overflow

Reply:
In nodejs, the maximum safe integer is 2<sup>53</sup>-1, while the total bitcoin supply is around 21 million, which is about 2<sup>50.899</sup>; and the change may affect current use of it, so at this time, we may not change it